### PR TITLE
CM-22208 Add JSON output format for scan errors

### PR DESCRIPTION
**Pull request by MarshalX**

### CODE REVIEW

File: cli/code_scanner.py [hunk 1]
```diff
@@ -818,33 +818,62 @@ def _is_subpath_of_cycode_configuration_folder(filename: str) -> bool:
...
+        InvalidGitRepositoryError: {
+            'soft_fail': False,
+            'code': 'invalid_git_error',
+            'message': 'The path you supplied does not correlate to a git repository. '
+                       'Should you still wish to scan this path, use: ‘cycode scan path <path>’'
+        },
+    }
+
+    # TODO(MarshalX): Extend functionality of CLI printers and move this
+    def print_error(error_data) -> None:
```
Comment (MichalBor) [inline cli/code_scanner.py:861]: we prefer to not create "internal" functions even if their only use is here. please move it and call it `_print_error`

File: cli/code_scanner.py [hunk 2]
```diff
@@ -818,33 +818,62 @@ def _is_subpath_of_cycode_configuration_folder(filename: str) -> bool:
...
+        if error_data['soft_fail']:
```
[the remark this answers is not in the document]
Comment (MichalBor) [reply cli/code_scanner.py:862]: notice that for `InvalidGitRepositoryError` we do not set soft_fail = true

[the remark this answers is not in the document]
Comment (MarshalX) [reply cli/code_scanner.py:862]: > notice that for `InvalidGitRepositoryError` we do not set soft_fail = true

thank you. should works fine 
image

[the remark this answers is not in the document]
Comment (MarshalX) [reply cli/code_scanner.py]: I double-checked it. it works

**Comment by MarshalX** [comment]
@MichalBor in my last commit I use `TypedDict`. It's available from Python 3.8 but I am a bit confused. The README file of this repository gives me this:
https://github.com/cycodehq-public/cycode-cli/blob/8937ae8f2a5ff656fa7f78d9c6a8d791a144d1d3/README.md?plain=1#L49
but the setup file gives this:
https://github.com/cycodehq-public/cycode-cli/blob/8937ae8f2a5ff656fa7f78d9c6a8d791a144d1d3/setup.py#L46

could you pls tell me what is the lowest version of python that CLI is support?

**Comment by MichalBor** [comment]
> @MichalBor in my last commit I use `TypedDict`. It's available from Python 3.8 but I am a bit confused. The README file of this repository gives me this:
> 
> https://github.com/cycodehq-public/cycode-cli/blob/8937ae8f2a5ff656fa7f78d9c6a8d791a144d1d3/README.md?plain=1#L49
> 
> 
> but the setup file gives this:
> https://github.com/cycodehq-public/cycode-cli/blob/8937ae8f2a5ff656fa7f78d9c6a8d791a144d1d3/setup.py#L46
> 
> could you pls tell me what is the lowest version of python that CLI is support?

ah, honestly I'm not sure. AFAIK we support python 3.7 as well. in general lets stick in avoiding writing a logic that is supported from specific version

**Comment by MichalBor** [comment]
@artem-fedorov @MarshalX regarding the q above and for better monitoring our users python version usage - lets add python_version param to the "report" scan result API. not urgent, open a ticket to the backlog please

**Comment by MarshalX** [comment]
> in general lets stick in avoiding writing a logic that is supported from specific version

@MichalBor unfortunately, it is not possible in the Python world( the project uses f-strings already, for example. f strings were added in Python 3.6 and so on. Some features were added in Python 3.7. Well, we need to specify the lowest version of Python for CLI and don't use features from higher versions. All higher versions should work fine  

I can do a small research to figure out if there are features in use from Python 3.7. I am 100% sure that at the current state of the project, it doesn't support Python lower than 3.6

**Comment by MichalBor** [comment]
@MarshalX, ok, but lets support 3.7 and above for now, because we don't know what the min python version that used by the users and in the setup.py we declare we support it. regarding python version 3.6 - yep we don't support it. I don't remember exactly why, but we got an error for that specific version and preferred not to fix it since we had not users with this version at that time.

**Review by MichalBor — APPROVED** [review]

**Comment by MarshalX** [comment]
@MichalBor I added tests for `soft_fail` to be 100% sure that everything works fine. I ran tests locally on the main branch and it fully covers `_handle_exception`. That's all changes that I did after the latest approve. Pls approve again

**Review by MichalBor — APPROVED** [review]
Well done!

**Comment by MarshalX** [comment]
re sign commits due to manipulation from GitHub side

**Pull request merged**